### PR TITLE
Adding support for missing request and response messages

### PR DESCRIPTION
**Pull request by campoy**

### CODE REVIEW

**Comment by campoy** [comment]
What do you think about this, @crhym3 ?

File: README.md [hunk 1]
```diff
@@ -88,6 +111,14 @@ func init() {
...
     "greets.list", "GET", "greetings", "List most recent greetings."
 
+  info = api.MethodByName("Add").Info()
+  info.Name, info.HTTPMethod, info.Path, info.Desc =
+    "greets.add", "PUT", "greetings", "Add a greeting."
+
+  info = api.MethodByName("Count").Info()
+  info.Name, info.HTTPMethod, info.Path, info.Desc =
+    "greets.count", "GET", "greetings/count", "Count all greetings."
+
```
Comment (crhym3) [inline README.md]: maybe also do the same thing here, define your neat `method()` closure? I liked it a lot.

File: endpoints/service.go [hunk 2]
```diff
@@ -24,6 +25,8 @@ var (
 	typeOfRequest = reflect.TypeOf((*http.Request)(nil)).Elem()
 	// Precompute the reflect type for Context.
 	typeOfContext = reflect.TypeOf((*Context)(nil)).Elem()
+	// Precompute the reflect type for *VoidMessage.
+	typeOfVoidMessage = reflect.TypeOf(new(VoidMessage))
```
Comment (crhym3) [inline endpoints/service.go]: Is it possible to do something like this?
typeOfVoidMessage = reflect.TypeOf((*VoidMessage)(nil))

**Comment by crhym3** [comment]
Sounds good! But should also modify tests... I know, don't tell me about it :)

File: endpoints/service.go (hunk 2 above)
Comment (campoy) [inline endpoints/service.go]: No, if we do that it panics when we call Interface, since it's a zero value.

File: README.md (hunk 1 above)
Comment (campoy) [inline README.md]: done

File: endpoints/service.go (hunk 2 above)
Comment (crhym3) [inline endpoints/service.go]: Ah yeah, because we need VoidMessage's Interface. Got it.

**Comment by crhym3** [comment]
I could modify tests to work with this but not today - have a couple of things to do first.

**Comment by campoy** [comment]
tests added and passing

File: endpoints/server_test.go [hunk 3]
```diff
@@ -86,6 +86,27 @@ func (s *ServerTestService) MsgWithReturn(c Context, req *TestMsg) (*TestMsg, er
 	return &TestMsg{req.Name}, nil
 }
 
+func (s *ServerTestService) MsgWithoutRequest(c Context) (*TestMsg, error) {
+	if c == nil {
+		return nil, errors.New("MsgReturnResp: c = nil")
```
Comment (crhym3) [inline endpoints/server_test.go]: `MsgWithoutRequest` for consistency.

File: endpoints/server_test.go [hunk 4]
```diff
@@ -86,6 +86,27 @@ func (s *ServerTestService) MsgWithReturn(c Context, req *TestMsg) (*TestMsg, er
...
+	}
+	return &TestMsg{}, nil
+}
+
+func (s *ServerTestService) MsgWithoutRespose(c Context, req *TestMsg) error {
+	if c == nil {
+		return errors.New("MsgReturnResp: c = nil")
```
Comment (crhym3) [inline endpoints/server_test.go]: `MsgWithoutRespose`

File: endpoints/server_test.go [hunk 5]
```diff
@@ -86,6 +86,27 @@ func (s *ServerTestService) MsgWithReturn(c Context, req *TestMsg) (*TestMsg, er
...
+	}
+	return nil
+}
+
+func (s *ServerTestService) MsgWithoutRequestNorRespose(c Context) error {
+	if c == nil {
+		return errors.New("MsgReturnResp: c = nil")
```
Comment (crhym3) [inline endpoints/server_test.go]: `MsgWithoutRequestNorRespose`

**Comment by campoy** [comment]
All of the above done.

**Comment by crhym3** [comment]
LGTM!

**Pull request merged**